### PR TITLE
fix(dynamic-forms): removed regex for name

### PR DESCRIPTION
**Pull request by JulieMarie**
## Description
Removed regex from name string

### Test Steps
- [ ] checkout branch
- [ ] `npm run serve`

### General Tests for Every PR

- [ ] `npm run serve:prod` still works.
- [ ] `npm run tslint` passes.
- [ ] `npm run stylelint` passes.
- [ ] `npm test` passes and code coverage is not lower.
- [ ] `npm run build:lib` still works.

##### Stackblitz with regex implemented
https://stackblitz.com/edit/ajc-stackoverflow-48705028?file=app/app.component.ts
covalent-fix

### CODE REVIEW

File: src/platform/dynamic-forms/services/dynamic-forms.service.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export interface ITdDynamicElementConfig {
   validators?: ITdDynamicElementValidator[];
 }
 
-export const DYNAMIC_ELEMENT_NAME_REGEX: RegExp = /^[a-zA-Z]+[a-zA-Z0-9-_]*$/;
+export const DYNAMIC_ELEMENT_NAME_REGEX: RegExp = /^[^0-9][^\@]*$/;
```
Comment (ilsiepotamus) [inline src/platform/dynamic-forms/services/dynamic-forms.service.ts]: what are you trying to capture with this regex?

Comment (JulieMarie) [reply src/platform/dynamic-forms/services/dynamic-forms.service.ts]: In the original unit tests, it's testing for a name starting with a number or a name with an ampersand in it.  I do realize that the `\` isn't necessary to escape the `@`.

Comment (ilsiepotamus) [reply src/platform/dynamic-forms/services/dynamic-forms.service.ts]: ok, so it actually is testing for what you want to be testing for, however.  I just thought it was an odd pattern.

Comment (JulieMarie) [reply src/platform/dynamic-forms/services/dynamic-forms.service.ts]: Agreed.

**Pull request merged**